src/analysis/max_min_driving_force.jl

```diff
@@ -131,8 +171,164 @@ function max_min_driving_force(
+[`max_min_driving_force`](@ref), and `relax_driving_force` times the optimum. If
+`relax_driving_force = 1`, then the max min driving force of the problem is fixed to that
+value instead of constrained.
+
+Returns a matrix of solutions to [`max_min_driving_force`](@ref) additionally constrained as
+described above, where the rows are in the order of the reactions and then the metabolites
+of the `model`. For the reaction rows the first column is the maximum dG of that reaction,
+and the second column is the minimum dG of that reaction subject to the above constraints.
+For the metabolite rows, the first column is the maximum concentration, and the second column
+is the minimum concentration subject to the constraints above.
+"""
+function max_min_driving_force_variability(
+    model::MetabolicModel,
+    reaction_standard_gibbs_free_energies::Dict{String,Float64},
+    optimizer;
+    workers = [myid()],
+    relax_driving_force = 0.9,
+    flux_solution::Dict{String,Float64} = Dict{String,Float64}(),
+    proton_ids::Vector{String} = ["h_c", "h_e"],
+    water_ids::Vector{String} = ["h2o_c", "h2o_e"],
+    constant_concentrations::Dict{String,Float64} = Dict{String,Float64}(),
+    concentration_ratios::Dict{Tuple{String,String},Float64} = Dict{
+        Tuple{String,String},
+        Float64,
+    }(),
+    concentration_lb = 1e-9,
+    concentration_ub = 100e-3,
+    T = _constants.T,
+    R = _constants.R,
+    small_flux_tol = 1e-6,
+    modifications = [],
+    ignore_reaction_ids = [],
+)
+    initsol = max_min_driving_force(
+        model,
+        reaction_standard_gibbs_free_energies,
+        optimizer;
+        flux_solution,
+        proton_ids,
+        water_ids,
+        constant_concentrations,
+        concentration_ratios,
+        concentration_lb,
+        concentration_ub,
+        T,
+        R,
+        small_flux_tol,
+        modifications,
+        ignore_reaction_ids,
+    )
+
+    mmdf = initsol.mmdf
+
+    dgr_variants = [
+        [[_mmdf_add_df_bound(mmdf, relax_driving_force), _mmdf_dgr_objective(ridx, sense)]] for ridx = 1:n_reactions(model), sense in [MOI.MAX_SENSE, MOI.MIN_SENSE]
+    ]
+    concen_variants = [
+        [
+            [
+                _mmdf_add_df_bound(mmdf, relax_driving_force),
+                _mmdf_concen_objective(midx, sense),
+            ],
+        ] for midx = 1:n_metabolites(model), sense in [MOI.MAX_SENSE, MOI.MIN_SENSE]
+    ]
+
+    return screen(
+        model;
+        args = [dgr_variants; concen_variants],
+        analysis = (m, args) -> max_min_driving_force(
+            m,
+            reaction_standard_gibbs_free_energies,
+            optimizer;
+            flux_solution,
+            proton_ids,
+            water_ids,
+            constant_concentrations,
+            concentration_ratios,
+            concentration_lb,
+            concentration_ub,
+            T,
+            R,
+            small_flux_tol,
+            modifications = [modifications; args],
```

Comment:
how does this work? (likely wrong)

Reply:
ah you are forwarding functions, ok good. :D

Reply:
possibly better to change the order again though, first args, then modifications. User could have made the modifications on the original model themselves.

Reply:
fixing!